STEP/CodeGeneration/StandardEnvironmentVisitor.cs

```diff
@@ -0,0 +1,307 @@
+    {
+        n.Left.Accept(this);
+    }
+
+    public void Visit(WhileNode n)
+    {
+        n.Condition.Accept(this);
+        foreach(var stmt in n.Body)
+        {
+            stmt.Accept(this);
+        }
+    }
+
+    public void Visit(ForNode n)
+    {
+        n.Initializer.Accept(this);
+        n.Update.Accept(this);
+        n.Limit.Accept(this);
+        foreach(var stmt in n.Body)
+        {
+            stmt.Accept(this);
+        }
+    }
+
+    public void Visit(ContNode n)
+    {
+        
+    }
+
+    public void Visit(BreakNode n)
+    {
+        
+    }
+
+    public void Visit(LoopNode n)
+    {
+        foreach(var stmt in n.Stmts)
+        {
+            stmt.Accept(this);
+        }
+    }
+
+    public void Visit(FuncDefNode n)
+    {
+        foreach(var stmt in n.Stmts)
+        {
+            stmt.Accept(this);
+        }
+    }
+
+    public void Visit(FuncExprNode n)
+    {
+        // Check for Arduino standard function
+        if (n.Id.AttributesRef is StdFuncSymTableEntry symbol)
+        {
+            n.Id.Id = symbol.ArduinoName;
+            foreach(var param in n.Params)
+            {
+                param.Accept(this);
+            }
+        }
+    }
+
+    public void Visit(FuncStmtNode n)
+    {
+        // Check for Arduino standard function
+        if (n.Id.AttributesRef is StdFuncSymTableEntry symbol)
+        {
+            n.Id.Id = symbol.ArduinoName;
+            foreach (var param in n.Params)
+            {
+                param.Accept(this);
+            }
+        }
+    }
+
+    public void Visit(FuncsNode n)
+    {
+        foreach(var dcl in n.FuncDcls)
+        {
+            dcl.Accept(this);
+        }
+    }
+
+    public void Visit(RetNode n)
+    {
+        
+    }
+
+    public void Visit(IfNode n)
+    {
+        n.Condition.Accept(this);
+        foreach(var stmt in n.ThenClause)
+        {
+            stmt.Accept(this);
+        }
+        if(n.ElseIfClauses != null)
+        {
+            foreach (var elseIf in n.ElseIfClauses)
+            {
+                elseIf.Accept(this);
+            }
+        }
+        if (n.ElseClause != null)
+        {
+            foreach (var stmt in n.ElseClause)
+            {
+                stmt.Accept(this);
+            }
+        }
+    }
+
+    public void Visit(ElseIfNode n)
+    {
+        n.Condition.Accept(this);
+        foreach (var stmt in n.Body)
+        {
+            stmt.Accept(this);
+        }
+    }
+
+    public void Visit(VarsNode n)
+    {
+        foreach(var dcl in n.Dcls)
+        {
+            dcl.Accept(this);
+        }
+    }
+
+    public void Visit(ProgNode n)
+    {
+        n.VarsBlock?.Accept(this);
+        n.FuncsBlock?.Accept(this);
+        n.SetupBlock?.Accept(this);
+        n.LoopBlock?.Accept(this);
+    }
+
+    public void Visit(SetupNode n)
+    {
+        foreach(var stmt in n.Stmts)
+        {
+            stmt.Accept(this);
+        }
+    }
+}
+
```

Comment:
Wow hvor ville det have været smart hvis vi havde lavet en base visitor som havde hele traversal logikken indbygget, og så bare havde custom processing per node - så man ikke skulle skrive hele det her visit skelet for hver ny visitor man laver 🤓